trainer.py

```diff
@@ -0,0 +1,42 @@
+"""
+Command line interface to run the neural network model!
+
+From the project root directory, do:
+
+    python trainer.py fit
+
+References:
+- https://lightning.ai/docs/pytorch/2.1.0/cli/lightning_cli.html
+- https://pytorch-lightning.medium.com/introducing-lightningcli-v2-supercharge-your-training-c070d43c7dd6
+"""
+import torch
+from lightning.pytorch.cli import ArgsType, LightningCLI
+
+from src.datamodule import BaseDataModule
+from src.model import BaseLitModule
+
+
+# %%
+def cli_main(
+    save_config_callback=None,
+    seed_everything_default=42,
+    trainer_defaults: dict = {"logger": False, "precision": "bf16-mixed"},
+    args: ArgsType = None,
+):
+    """
+    Command-line inteface to run BaseLitModule with BaseDataModule.
+    """
+    cli = LightningCLI(
+        model_class=BaseLitModule,
+        datamodule_class=BaseDataModule,
+        save_config_callback=save_config_callback,
+        seed_everything_default=seed_everything_default,
+        trainer_defaults=trainer_defaults,
+        args=args,
+    )
```

Comment:
Open to discuss on whether LightningCLI is ok to use, or if we want to go with something a bit more customizable like Hydra (note: not exactly apples-to-apples comparison, see e.g. https://github.com/ashleve/lightning-hydra-template/issues/443#issuecomment-1249114069).

We could also stick with LightningCLI for now (since it's built-in to Lightning), and move on to something else later when there's a need.

Reply:
As you suggested, let us start with `lightning-cli` & if we feel constrained anytime during the project, we can switch to `hydra`.